app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java

```diff
@@ -54,13 +57,14 @@
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        this.imm = (InputMethodManager) getActivity().getSystemService(Context.INPUT_METHOD_SERVICE);
         setHasOptionsMenu(true);
     }
 
     @Override
     public void onResume() {
         super.onResume();
-
+        getActivity().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
```

Comment:
Would there be any adverse affects if we specified this in the XML instead? e.g. in MainActivity, we have:
```xml
android:windowSoftInputMode="adjustResize|stateAlwaysHidden"
```

Could we include stateAlwaysVisible in SettingsActivity?

Reply:
To be honnest, that's the only workaround i found for onResume.. It seems like anything you do with InputMethodManager in onResume is simply ignored.

Reply:
Sorry i didn't even answered your question. It's just my preference to use java instead of xml for everything else than "core" UI elements. I can try to use the xml statement you quoted if u want me to :) . But if we put this in SettingsActivity, isn't the keyboard going to show up everywhere in the SettingsActivity, even if no keyboard input is necessary ?

Reply:
> It's just my preference to use java instead of xml for everything else than "core" UI elements. I can try to use the xml statement you quoted if u want me to :) .

Please do! We'd prefer XML because there's really only one place to change this attribute whereas it's harder to track down a single line of code.

> But if we put this in SettingsActivity, isn't the keyboard going to show up everywhere in the SettingsActivity, even if no keyboard input is necessary ?

My understanding of this attribute is that it will take affect only when keyboard input is necessary (i.e. it won't show a keyboard when the activity is shown if there's no input on the screen). But please correct me if I'm wrong!

Reply:
Hey @mcomella ! I tried with an XML statement as u quoted above, but i'm afraid it's not working. Because if we add this statement in the manifest for SettingsActivity, the keyboard shows up at the start of the activity, even tho there's no input field on screen). The problem is that ManualAddSearchEngine is a fragment and not an activity itself. Therefore, i think we don't have the choice, we have to do it programatically. Any suggestions or ideas ? Please let me know.